cpp/src/arrow/python/builtin_convert.cc

```diff
@@ -390,28 +390,26 @@ class TypedConverterVisitor : public TypedConverter<BuilderType> {
     }
     return Status::OK();
   }
-
-  virtual Status AppendItem(const OwnedRef& item) = 0;
 };
 
 class NullConverter : public TypedConverterVisitor<NullBuilder, NullConverter> {
  public:
-  inline Status AppendItem(const OwnedRef& item) {
+  Status AppendItem(const OwnedRef& item) {
```

Comment:
I'm not sure why these were ever virtual. Making them non-virtual because we're already using CRTP here

Reply:
Thanks, sounds good to me